libdevcore/CommonData.cpp

```diff
@@ -86,20 +86,27 @@ bool dev::passesAddressChecksum(string const& _str, bool _strict)
 	))
 		return true;
 
+	return _str == dev::getChecksummedAddress(_str);
+}
+
+string dev::getChecksummedAddress(string const& _addr)
+{
+	string s = _addr.substr(0, 2) == "0x" ? _addr.substr(2) : _addr;
 	h256 hash = keccak256(boost::algorithm::to_lower_copy(s, std::locale::classic()));
+
+	assertThrow(s.length == 40, "");
+	assertThrow(hash.find_first_not_of("abcdef") == string::npos, "");
```

Comment:
`hash` is a `h256`, i.e. it is binary data. I don't think we should search for hex digits there.

Reply:
Yeh it should be done before on the input.